File: java/air-io-network/src/main/java/org/feuyeux/air/io/network/udpctrl/controlinfo/codec/KeyControlInfoCodec.java
package org.feuyeux.air.io.network.udpctrl.controlinfo.codec;

import org.feuyeux.air.io.network.udpctrl.controlinfo.ControlInfo;
import org.feuyeux.air.io.network.udpctrl.controlinfo.KeyControlInfo;

/**
 * Created by erichan on 2/10/14.
 */
public class KeyControlInfoCodec implements ControlInfoCodec {
    KeyControlInfoCodec() {
    }

    @Override
    public byte[] encode(ControlInfo controlInfo) {
        KeyControlInfo keyControlInfo = (KeyControlInfo) controlInfo;
        return keyControlInfo.getKeyPress().getBytes();
    }

    @Override
    public ControlInfo decode(byte[] buf) {
        KeyControlInfo keyControlInfo = new KeyControlInfo();
        keyControlInfo.setKeyPress(new String(buf));
        return keyControlInfo;
    }
}

File: ios/learning/UdpCmd/UdpCmd/KeyControlInfoCodec.m
//
//  KeyControlInfoCodec.m
//  UdpCmd
//
//  Created by feuyeux@gmail.com on 14-2-13.
//  Copyright (c) 2014年 feuyeux@gmail.com. All rights reserved.
//

#import "KeyControlInfoCodec.h"
#import "KeyControlInfo.h"

@implementation KeyControlInfoCodec

-(NSData *) encode:( ControlInfo  *)controlInfo{
    if ([controlInfo isKindOfClass:[KeyControlInfo class]])
    {
        KeyControlInfo *keyControlInfo = (KeyControlInfo *)controlInfo;
        NSData *data= [keyControlInfo.keyPress dataUsingEncoding:NSUTF8StringEncoding];
        return data;
    }else{
        return NULL;
    }
}

-(ControlInfo *) decode:(NSData *)data{
    NSString* keyPress = [NSString stringWithUTF8String:[data bytes]];
    KeyControlInfo *keyControlInfo = [[KeyControlInfo alloc] initWithKeyPress:keyPress];
    return keyControlInfo;
}

@end
